cmd/cloudFoundryDeploy.go

```diff
@@ -0,0 +1,855 @@
+	if len(manifestVariablesFiles) == 1 && manifestVariablesFiles[0] == defaultManifestVariableFileName {
+		// we have only the default file. Most likely this is not configured, but we simply have the default.
+		// In case this file does not exist we ignore that file.
+		exists, err := fileUtils.FileExists(defaultManifestVariableFileName)
+		if err != nil {
+			return []string{}, err
+		}
+		if !exists {
+			return []string{}, nil
+		}
+	}
+	return manifestVariablesFiles, nil
+}
+
+func toParameterMap(parameters []string) (*orderedmap.OrderedMap, error) {
+
+	parameterMap := orderedmap.NewOrderedMap()
+
+	for _, p := range parameters {
+		keyVal := strings.Split(p, "=")
+		if len(keyVal) != 2 {
+			return nil, fmt.Errorf("Invalid parameter provided (expected format <key>=<val>: '%s'", p)
+		}
+		parameterMap.Set(keyVal[0], keyVal[1])
+	}
+	return parameterMap, nil
+}
+
+func handleLegacyCfManifest(manifestFile string) error {
+	manifest, err := _getManifest(manifestFile)
+	if err != nil {
+		return err
+	}
+
+	err = manifest.Transform()
+	if err != nil {
+		return err
+	}
+	if manifest.IsModified() {
+
+		err = manifest.WriteManifest()
+
+		if err != nil {
+			return err
+		}
+		log.Entry().Infof("Manifest file '%s' was in legacy format has been transformed and updated.", manifestFile)
+	} else {
+		log.Entry().Infof("Manifest file '%s' was not in legacy format. No tranformation needed, no update performed.", manifestFile)
+	}
+	return nil
+}
+
+func prepareCfPushCfNativeDeploy(config *cloudFoundryDeployOptions) (string, []string, []string, error) {
+
+	deployOptions := []string{}
+	varOptions, err := getVarOptions(config.ManifestVariables)
+	if err != nil {
+		return "", []string{}, []string{}, err
+	}
+	varFileOptions, err := getVarFileOptions(config.ManifestVariablesFiles)
+	if err != nil {
+		return "", []string{}, []string{}, err
+	}
+
+	deployOptions = append(deployOptions, varOptions...)
+	deployOptions = append(deployOptions, varFileOptions...)
+
+	return "push", deployOptions, []string{}, nil
+}
+
+func toStringInterfaceMap(in *orderedmap.OrderedMap, err error) (map[string]interface{}, error) {
+
+	out := map[string]interface{}{}
+
+	if err == nil {
+		for _, key := range in.Keys() {
+			if k, ok := key.(string); ok {
+				val, exists := in.Get(key)
+				if exists {
+					out[k] = val
+				} else {
+					return map[string]interface{}{}, fmt.Errorf("No entry found for '%v'", key)
+				}
+			} else {
+				return map[string]interface{}{}, fmt.Errorf("Cannot cast key '%v' to string", key)
+			}
+		}
+	}
+
+	return out, err
+}
+func getVarOptions(vars []string) ([]string, error) {
+
+	varsMap, err := toParameterMap(vars)
+	if err != nil {
+		return []string{}, err
+	}
+
+	varsResult := []string{}
+
+	for _, key := range varsMap.Keys() {
+		val, _ := varsMap.Get(key)
+		if v, ok := val.(string); ok {
+			varsResult = append(varsResult, "--var", fmt.Sprintf("%s=%s", key, quoteAndBashEscape(v)))
```

Comment:
I think it may be wrong to quote and escape here. Or will this be written to a script?

Reply:
Full ACK. There is no shell inbetween which consumes the quotes and the escapes. --> adjusted